samcli/lib/providers/cfn_api_provider.py

```diff
@@ -75,13 +72,14 @@ def extract_resources(self, stacks: List[Stack], collector, cwd=None):
                 if resource_type == CfnApiProvider.APIGATEWAY_V2_STAGE:
                     self._extract_cfn_gateway_v2_stage(resources, resource, collector)
 
-        all_apis = []
-        for _, apis in collector:
-            all_apis.extend(apis)
-        return all_apis
```

Comment:
we never used the return value of `extract_resources()`, removing the return value makes this method consistent with the ones in the super class and another subclass.

Reply:
returning the extracted resources here and in the other methods as well makes more sense to me, but it is fine if you are sure this won't break anything

Reply:
Yeah, I think so too. Especially in function provider and layer provider, `extract_*()` returns the extraction results. 
In this PR, I don't want to change too much behaviors so I only remove unused code. 

It does not break anything:

https://github.com/aws/aws-sam-cli/search?q=extract_resources

<img width="1055" alt="image" src="https://user-images.githubusercontent.com/3804518/110703357-87564980-81a8-11eb-85b7-0363161a2b27.png">